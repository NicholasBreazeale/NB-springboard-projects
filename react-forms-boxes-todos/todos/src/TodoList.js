import React, { useState } from "react";
import { v4 as uuidv4 } from "uuid";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

function TodoList() {
  const [todos, setTodos] = useState([]);

  function addTodo({ text }) {
    setTodos(t => [...t, {id: uuidv4(), text}]);
  }

  function removeTodo(id) {
    setTodos(todos.filter(t => t.id !== id));
  }

  return (
    <>
      <NewTodoForm addTodo={addTodo} />
      <div data-testid="todo-list">
        {todos.map(t =>
          <Todo text={t.text} removeTodo={() => removeTodo(t.id)} key={t.id} />
        )}
      </div>
    </>
  );
}

export default TodoList;
