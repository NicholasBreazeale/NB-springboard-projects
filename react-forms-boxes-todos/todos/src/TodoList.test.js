import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function() {
  render(<TodoList />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add and remove tasks", function() {
  const { queryByLabelText, queryByTestId, asFragment } = render(<TodoList />);

  // no tasks should be rendered yet
  const todoList = queryByTestId("todo-list");
  expect(asFragment()).toMatchSnapshot();

  const textInput = queryByLabelText("Todo:");
  const submitButton = queryByTestId("submit-btn");

  // create a new task
  fireEvent.change(textInput, { target: { value: "Hello world!" } });
  fireEvent.click(submitButton);

  // 1 task and its delete button should be rendered
  expect(asFragment()).toMatchSnapshot();

  // delete task and its button
  fireEvent.click(todoList.children[0]);
  expect(asFragment()).toMatchSnapshot();
});