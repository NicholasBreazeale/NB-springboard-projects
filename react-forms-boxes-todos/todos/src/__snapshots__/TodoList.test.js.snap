// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add and remove tasks 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Todo:
    </label>
    <input
      id="text"
      name="text"
      type="text"
      value=""
    />
    <button
      data-testid="submit-btn"
    >
      Add
    </button>
  </form>
  <div
    data-testid="todo-list"
  />
</DocumentFragment>
`;

exports[`can add and remove tasks 2`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Todo:
    </label>
    <input
      id="text"
      name="text"
      type="text"
      value=""
    />
    <button
      data-testid="submit-btn"
    >
      Add
    </button>
  </form>
  <div
    data-testid="todo-list"
  >
    <div>
      Hello world!
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`can add and remove tasks 3`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Todo:
    </label>
    <input
      id="text"
      name="text"
      type="text"
      value=""
    />
    <button
      data-testid="submit-btn"
    >
      Add
    </button>
  </form>
  <div
    data-testid="todo-list"
  >
    <div>
      Hello world!
      <button>
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="text"
    >
      Todo:
    </label>
    <input
      id="text"
      name="text"
      type="text"
      value=""
    />
    <button
      data-testid="submit-btn"
    >
      Add
    </button>
  </form>
  <div
    data-testid="todo-list"
  />
</DocumentFragment>
`;
